apps/src/lib/ui/SettingsCog.jsx

```diff
@@ -0,0 +1,105 @@
+/** @file Settings menu cog icon */
+import React, {Component} from 'react';
+import Radium from 'radium';
+import Portal from 'react-portal';
+import msg from '@cdo/locale';
+import FontAwesome from '../../templates/FontAwesome';
+import color from '../../util/color';
+import SettingsMenu from './SettingsMenu';
+
+const style = {
+  iconContainer: {
+    float: 'right',
+    marginRight: 10,
+    marginLeft: 10,
+    height: '100%',
+    cursor: 'pointer',
+    color: color.lighter_purple,
+    ':hover': {
+      color: color.white,
+    }
+  },
+  assetsIcon: {
+    fontSize: 18,
+    verticalAlign: 'middle',
+  },
+};
+
+class SettingsCog extends Component {
+  constructor(props) {
+    super(props);
+
+    // Autobind
+    this.open = this.open.bind(this);
+    this.beforeClose = this.beforeClose.bind(this);
+    this.close = this.close.bind(this);
+
+    // Default icon bounding rect for first render
+    this.targetPoint = {top: 0, left: 0};
+  }
+
+  // This ugly two-flag state is a workaround for an event-handling bug in
+  // react-portal that prevents closing the portal by clicking on the icon
+  // that opened it.  For now we're just disabling the cog when the menu is
+  // open, and re-enabling one tick after it closes.
+  // @see https://github.com/tajo/react-portal/issues/140
+  state = {
+    open: false,
+    canOpen: true,
+  };
+
+  open() {
+    this.setState({open: true, canOpen: false});
+  }
+
+  beforeClose(_, resetPortalState) {
+    resetPortalState();
+    this.setState({open: false});
+    window.setTimeout(() => this.setState({canOpen: true}), 0);
+  }
+
+  close() {
+    this.setState({open: false});
+  }
+
+  setTargetPoint(icon) {
+    if (!icon) {
+      return;
+    }
+
+    const rect = icon.getBoundingClientRect();
+    this.targetPoint = {
+      top: rect.bottom - 6,
+      left: rect.left + (rect.width / 2) - 1,
```

Comment:
Just experimented until it seemed about right.  The -6 vertical offset let's the menu's tail overlap the cog just a little bit.  The -1 horizontal offset is because it just looked wrong to me.